### ISSUE TITLE
Crash when deleting buffer after a nested autocommand gets buffer options using python

### ISSUE THREAD

**Issue by zeertzjq** [vim/vim#11631]
### Steps to reproduce

1. Run `vim --clean`
2. Source the following file:
```vim
set hidden
edit Xa
edit Xb
call setline(1, 'bar')
enew
au BufWriteCmd Xb ++nested call setbufvar('Xa', '&ft', 'foo') | bw Xa
au FileType foo call py3eval('vim.current.buffer.options["cindent"]')
wall
```
3. Crash

### Expected behaviour

No crash

### Version of Vim

9.0.910

### Environment

Operating system: Arch Linux
Terminal: kitty
Value of $TERM: xterm-kitty
Shell: fish

**Comment by zeertzjq** [comment]
I haven't managed to build Vim from source with if_pyth, but I think this is related to these lines in `aucmd_prepbuf()`:
```c
    if (win == NULL && aucmd_win_used)
	// Strange recursive autocommand, fall back to using the current
	// window.  Expect a few side effects...
	win = curwin;
```

**Comment by zeertzjq** [comment]
Fixed by patch 9.0.0965
